### DISCUSSION TITLE
spanner input

### DISCUSSION THREAD

**Discussion by madalina2311** [redpanda-data/connect#1328] [Ideas]
Hello. Is there a plan to support Cloud spanner as input ?

**Comment by mihaitodor** [comment]
I never used it before and don't know much about it, but, based on a cursory read, it looks compatible with the `"database/sql"` package in the standard library: https://github.com/googleapis/go-sql-spanner. They also seem to offer a Docker image for an emulator: https://cloud.google.com/spanner/docs/emulator#docker.

If the above assumptions are correct, then adding this should be as easy as importing `"github.com/googleapis/go-sql-spanner"` [here](https://github.com/benthosdev/benthos/blob/a4a8061d18f243ce44802e631e540190a86287b9/public/components/all/package.go#L34-L38) and adding integration tests [here](https://github.com/benthosdev/benthos/blob/4debec5bf3011b759e72fd194ad8d81a00554534/internal/impl/sql/integration_test.go#L537-L541). Feel free to send a PR if you'd like to try it out, but, if not, let me know and I might get some time to look into it.

**Comment by mihaitodor** [comment]
Spanner support was added in [v4.42.0](https://github.com/redpanda-data/connect/releases/tag/v4.42.0).
